code/src/cljs/sixsq/nuvla/ui/edge_detail/views.cljs

```diff
@@ -557,107 +471,600 @@
+               ^{:key id}
+               [ui/TableRow
+                [ui/TableCell [values/as-link id :label (general-utils/id->short-uuid id)]]
+                [ui/TableCell timestamp]
+                [ui/TableCell category]
+                [ui/TableCell (:state content)]])]])
+
+
+         [uix/Pagination {:totalPages   total-pages
+                          :activePage   @page
+                          :onPageChange (ui-callback/callback
+                                          :activePage #(do
+                                                         (dispatch [::events/set-page %])
+                                                         (refresh (:id @nuvlabox))))}]]))))
+
+
+(def vuln-critical-color "#f41906")
+(def vuln-high-color "#f66e0a")
+(def vuln-medium-color "#fbbc06")
+(def vuln-low-color "#21b802")
+(def vuln-unknown-color "#949494")
+
+
+; there's a similar function in edge.views which can maybe be generalized
+(defn VulnStatisticState
+  [value label label-popup color state-selector]
+  (let [selected?      (or
+                         (= label @state-selector)
+                         (and (= label "collected")
+                           (= @state-selector nil)))
+        selected-style    (if selected? {:color "black"
+                                         :font-weight "bold"
+                                         :padding   "10px"}
+                                        {:color color
+                                         :font-weight "normal"
+                                         :padding "3px"})]
+    [ui/Statistic {:style    {:cursor "pointer"}
+                   :on-click #(dispatch [::events/set-vuln-severity-selector
+                                         (if (= label "collected") nil label)])}
+     [ui/StatisticValue {:style selected-style}
+                         (or value "-")]
+     [ui/StatisticLabel [ui/Popup
+                         {:trigger        (r/as-element [:span {:style selected-style} label])
+                          :content        label-popup
+                          :position       "bottom center"
+                          :on             "hover"
+                          :size           "tiny"
+                          :hide-on-scroll true}]]]))
+
+
+(defn VulnerabilitiesTableBody
+  [vulnerability-id product vulnerability-score color]
+  [ui/TableRow
+   [ui/TableCell vulnerability-id]
+   [ui/TableCell product]
+   [ui/TableCell {:style {:background-color color
+                          :font-weight  "bold"}} vulnerability-score]])
+
+
+(defn TabVulnerabilities
+  []
+  (let [tr              (subscribe [::i18n-subs/tr])
+        nb-status       (subscribe [::subs/nuvlabox-status])
+        state-selector  (subscribe [::subs/vuln-severity-selector])]
+    (fn []
+      (let [vulns   (:vulnerabilities @nb-status)
+            summary   (:summary vulns)
+            items     (:items   vulns)
+            items-extended (into []
```

Comment:
This code should be moved to a reactive subscription. Like that the render function will be only about rendering